### ISSUE TITLE
Classify in SoftmaxRegression

### ISSUE THREAD

**Issue by micyril** [mlpack/mlpack#935]
The class SoftmaxRegression implements a classification algorithm, but its method for classification has the name "Predict" rather than "Classify"  (what is less conventional among mlpack classification algorithms):
```
  /**
   * Predict the class labels for the provided feature points. The function
   * calculates the probabilities for every class, given a data point. It then
   * chooses the class which has the highest probability among all.
   *
   * @param testData Matrix of data points for which predictions are to be made.
   * @param predictions Vector to store the predictions in.
   */
  void Predict(const arma::mat& testData, arma::Row<size_t>& predictions) const;
```
It seems to me that we should deprecate the method Predict and add a method Classify instead of it, like it was done for LogisticRegression:
```
  /**
   * Predict the responses to a given set of predictors.  The responses will be
   * either 0 or 1.  Optionally, specify the decision boundary; logistic
   * regression returns a value between 0 and 1.  If the value is greater than
   * the decision boundary, the response is taken to be 1; otherwise, it is 0.
   * By default the decision boundary is 0.5.
   *
   * This method is deprecated---you should use Classify() instead.
   *
   * @param predictors Input predictors.
   * @param responses Vector to put output predictions of responses into.
   * @param decisionBoundary Decision boundary (default 0.5).
   */
  void Predict(const MatType& predictors,
               arma::Row<size_t>& responses,
               const double decisionBoundary = 0.5) const;
```

**Comment by sagarbhathwar** [comment]
Hi, I am new to Open Source Contribution but have some experience in ML and C++. Is mlpack a good place to start? Should I take up this (issue) and make logistic regression and softmax regression have uniform interface?

Thanks

**Comment by rcurtin** [comment]
Hi Kirill,

I agree with this strategy, I think it would be a good change. If you want to open a PR if you've already made the changes feel free.

@sagarbhathwar: I assume Kirill is already going to work on this one, I guess he can let you know if it is open or not.

**Comment by micyril** [comment]
Hi Ryan,

I have sent a PR.

**Comment by rcurtin** [comment]
Solved when #940 was merged.
